CourseManager/Courses/Course 1 Build a SwiftUI app for iOS 15/Views/CourseView.swift

```diff
@@ -0,0 +1,227 @@
+        Button {
+            withAnimation(.closeCard) {
+                show.toggle()
+                model.showDetail.toggle()
+            }
+        } label: {
+            Image(systemName: "xmark")
+                .font(.body.weight(.bold))
+                .foregroundColor(.secondary)
+                .padding(8)
+                .background(.ultraThinMaterial, in: Circle())
+        }
+        .frame(maxWidth: .infinity, maxHeight: .infinity, alignment: .topTrailing)
+        .padding(30)
+        .ignoresSafeArea()
+    }
+
+    var overlayContent: some View {
+        VStack(alignment: .leading, spacing: 12) {
+            Text(course.title)
+                .font(.largeTitle.weight(.bold))
+                .matchedGeometryEffect(id: "title\(course.id)", in: namespace)
+                .frame(maxWidth: .infinity, alignment: .leading)
+            Text(course.subtitle.uppercased())
+                .font(.footnote.weight(.semibold))
+                .matchedGeometryEffect(id: "subtitle\(course.id)", in: namespace)
+            Text(course.text)
+                .font(.footnote)
+                .matchedGeometryEffect(id: "text\(course.id)", in: namespace)
+
+            Divider()
+                .opacity(appear[0] ? 1 : 0)
+            HStack {
+                Image("Avatar Default")
+                    .resizable()
+                    .frame(width: 26, height: 26)
+                    .cornerRadius(10)
+                    .padding(8)
+                    .background(.ultraThinMaterial, in: RoundedRectangle(cornerRadius: 18, style: .continuous))
+                    .strokeStyle(cornerRadius: 18)
+                Text("Taught by Meng To")
+                    .font(.footnote)
+            }
+            .opacity(appear[1] ? 1 : 0)
+        }
+        .padding(20)
+        .background(
+            Rectangle()
+                .fill(.ultraThinMaterial)
+                .mask(RoundedRectangle(cornerRadius: 30, style: .continuous))
+                .matchedGeometryEffect(id: "blur", in: namespace)
+        )
+        .offset(y: 250)
+        .padding(20)
+    }
+
+    var drag: some Gesture {
+        DragGesture(minimumDistance: 30, coordinateSpace: .local)
+            .onChanged { value in
+                guard value.translation.width > 0 else { return }
+                if value.startLocation.x < 100 {
+                    withAnimation {
+                        viewState = value.translation
+                    }
+                }
+                if viewState.width > 120 {
+                    close()
+                }
+            }
+            .onEnded { _ in
+                if viewState.width > 80 {
+                    close()
+                } else {
+                    withAnimation {
+                        viewState = .zero
+                    }
+                }
+            }
+    }
+
+    func fadeIn() {
+        withAnimation(.easeOut.delay(0.3)) {
+            appear[0] = true
```

Comment:
improve logic with array.forEach()

Reply:
This didn't work, I'm guessing its because my View is a struct and so therefore I get the error Cannot assign to value: 'value' is a 'let' constant. I'm sure there is a better way to do this will think about it and update

Reply:
How is this as an alternative? 

func fadeIn() {
  var delay = 0.3
                for i in 0..<appear.count {
            
                withAnimation(.easeOut.delay(delay)) {
                    appear[i] = true
                    delay += 0.1
}
}